### PR TITLE
Linkify Roadmap.md

### PR DESCRIPTION
**Pull request by jornh**
### Summary of Changes

I think it’s useful to be able to place deep links to items on the roadmap like https://github.com/jornh/amundsen/blob/patch-2/docs/roadmap.md#native-lineage-integration

Sadly enough it’s a bit less visually pleasing as `<br>`s are not respected in headings - but I still think the benefit outweighs that.

*Also, more as a point for discussion:* How about we add links to issues to start capturing discussion/design and work coordination of some of the roadmap items? It would allow to gradually break down features into smaller “bite size” parts that people in the community might pick up and start implementing. The idea is basically to funnel people from seeing and wanting a feature to actually being able to more easily be able to start contribute (or am I just overthinking all this? 😜)

Cc: @philippemizrahi @markgrover

### Documentation

Oh yes! (Isn’t it always in this repo?)

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely.
- [x] PR includes a summary of changes.
- [x] I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -7,78 +7,78 @@ The following roadmap gives an overview of what we are currently working on and
 
 ## Current focus
 
-**Search & Resource page redesign**<br/>
+#### **Search & Resource page redesign**<br/>
```
Comment (feng-tao) [inline docs/roadmap.md]: do we still need <br/> here?

Comment (jornh) [reply docs/roadmap.md]: No, will clean it up. We also don't need to use ** to bold format as bold

Comment (feng-tao) [reply docs/roadmap.md]: still here?

File: docs/roadmap.md [hunk 2]
```diff
@@ -7,78 +7,114 @@ The following roadmap gives an overview of what we are currently working on and
...
+<br/>
 
 ## Next steps
 
-**Index Dashboards**
+#### Index Dashboards
 *What*: We want to help with the discovery of existing analysis work, dashboards. This is going to help avoid reinventing the wheel, create value for less technical users and help give context on how tables are used.<br/>
 *Status*: Product + technical specs are ready, designs are ready, implementation has not started<br/>
 *Links*: [Designs](https://drive.google.com/drive/folders/12oBrcXUsDtOsuU_QvO93LTvs4Dehx6az?usp=sharing) | [Product Specifications](https://docs.google.com/document/d/16cSKgM2sCYvhKq54yfwaHKwslJEGtdS2g5dcPV4p5qo/edit?usp=sharing) | [Technical RFC](https://docs.google.com/document/d/1PHk8OjcIULJ7hG0ckeMrRfTk3vXqnq5asEykgQUw-Ow/edit?usp=sharing)
+<br/>
```
Comment (feng-tao) [inline docs/roadmap.md]: and I don't think we need </br> for a line break in markdown file? we only need a space which should do the same.

Comment (jornh) [reply docs/roadmap.md]: In general you’re absolutely right. I did it here to add some extra white space after the short paragraphs to visually have the headings closer to the paragraph they head.

If you still prefer it without BRs I will change it.

Comment (jornh) [reply docs/roadmap.md]: Duh - had to re-read what you wrote ... and you're right. Though it has to be trailing double-spaces

**Pull request merged**